src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -571,16 +573,26 @@ DeferErrorIfCannotPushdownSubquery(Query *subqueryTree, bool outerQueryHasLimit)
 		errorDetail = "Offset clause is currently unsupported";
 	}
 
-	if (subqueryTree->limitCount && !outerQueryHasLimit)
+	/*
+	 * Limit is partially supported when SubqueryPushdown is set.
+	 * The outermost query must have a limit clause.
+	 */
+	if (subqueryTree->limitCount && !SubqueryPushdown)
```

Comment:
> implicit due to else if

I think that confused me a bit. Can't we make each as a separate `if` statements as we have throughout the function?

```C
	if (subqueryTree->limitCount && !SubqueryPushdown)
	{
		preconditionsSatisfied = false;
		errorDetail = "Limit in subquery is currently unsupported";
	}
	 
    if (subqueryTree->limitCount && SubqueryPushdown && !outerMostQueryHasLimit)
	{
		preconditionsSatisfied = false;
		errorDetail = "Limit in subquery without limit in the outermost query is "
					  "unsupported";
	}


```